consensus/doc/PROTOCOL_STATE_SYNC.md

```diff
@@ -0,0 +1,119 @@
+  ...
+}
+```
+Using the `PeerSyncAggregate`, a Node is able to compare its local `SyncState` against that of the Global Network.
+
+## State Sync Operation Modes
+
+State sync can be viewed as a state machine that transverses various modes the node can be in, including:
+* Synced Mode
+* Sync Mode
+* Server Mode
+
+The functionality of the node depends on the mode it is operating it. 
+
+*NOTE: that the modes are not necessarily mutually exclusive (e.g. the node can be in `Server Mode` and `Synced Mode` at the same time).*
+
+### Synced Mode
+If the Node is `Synced` or `localSyncState.Height == GlobalSyncMeta.Height` then the `StateSync` protocol is in `SyncedMode`.
+
+In `SyncedMode`, the Node is caught up to the latest block and relies on new blocks to be propagated via the P2P network after Validators reach consensus.
+
+### Sync Mode
+If the Node is `Syncing` or `localSyncState.Height < GlobalSyncMeta.Height` then the `StateSync` protocol is in `SyncMode`.
+
+In `SyncMode`, the Node is catching up to the latest block by making `BlockRequests` to its fellow eligible peers. A peer is eligible for a `BlockRequest` if `PeerMeta.MinHeight` <= `self.MaxBlockHeight` <= `PeerMeta.MaxHeight`.
+
+Though it is `unspecified` whether or not a Node may make `BlockRequests` in order or in parallel, the cryptographic restraints of block processing require the Node to call `ApplyingBlock` sequentially until it is `Synced`.
+
+It is important to note, if any blocks processed result in an invalid `AppHash` during `ApplyBlock`, a new `BlockRequest` must be issued until a valid block is found.
+
+### Server Mode
+
+In the `StateSync` protocol, the Node fields valid `BlockRequests` from its peers to help them `CatchUp` to be `Synced`. This sub-protocol is continuous throughout the lifecycle of StateSync.
+
+```mermaid
+graph TD
+    A[StateSync] -->|IsCaughtUp| B(Synced Mode)
+    B --> |Consensus Messages| C(ConsensusModule.Synced)
+    A --> |IsSyncing| E(Sync Mode)
+    E --> |Request Block| G[Peers]
+    G--> |ApplyBlock| A
+    A --> D[Server Mode]
+    D --> |Serve Blocks Upon Request| G
+```
+
+## Follow up tasks
+
+* `Fast Sync Design` - Sync only the last `N` blocks from `Latest Network Height`
+
+* `Optimistic Sync Design` - Optimize the State Sync protocol by parallelling requests and computation
+
+* `Block Chunk Design` - Update the Block by Block design to be able to receive and provide multiple blocks at a time.
+
+* `Block Stream Design` - Update the Block by Block design to stream blocks via a WebSocket from a single connectable peer.
+
+## Research Items
+TODO(M5): Create issues to track and discuss these work items.
+
+How the persistence layer design of `pruning` the Merkle Tree affects `StateSync`.
+
+How DB Pruning of the `SQL DB` might affect fast sync.
+
+How the Churn Management operations of RainTree might provide opportunities and obstacles with StateSync.
+
+## Glossary
```

Comment:
Question: do we need to differentiate between protocol-level actors and simply full nodes?

In the context of RainTree, I know we've discussed that list, being the same across all actors, is a function of the world state. For state sync, I believe full nodes are fair game as well.

I think adding a comment about this could be a helpful distinction.

Reply:
I think we do not at this level - but not strong on this. If you think it's helpful - add a comment before the merge